### PR TITLE
Commerce 7311 As a store admin, I want an order template that’s designed to be printed

### PR DESCRIPTION
**Pull request by christian-chiappa**

### CODE REVIEW

**Comment by christian-chiappa** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-report-impl/src/main/java/com/liferay/commerce/report/internal/exporter/CommerceReportExporterImpl.java [hunk 1]
```diff
@@ -43,13 +46,26 @@
...
 
-		try (InputStream inputStream = clazz.getResourceAsStream(
-				"dependencies/commerce_order.jrxml")) {
+		try {
+			InputStream inputStream = clazz.getResourceAsStream(
+				"dependencies/commerce_order.jrxml");
+
+			if (fileEntry != null) {
+				inputStream = fileEntry.getContentStream();
+			}
```
Comment (ivicac) [inline modules/apps/commerce/commerce-report-impl/src/main/java/com/liferay/commerce/report/internal/exporter/CommerceReportExporterImpl.java:69]: @christian-chiappa you mast close inputStream stream in the finally block

**Comment by christian-chiappa** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-channel-web/src/main/java/com/liferay/commerce/channel/web/internal/portlet/action/EditCommerceChannelMVCActionCommand.java [hunk 2]
```diff
@@ -81,15 +92,91 @@ protected void doProcessAction(
...
+		}
+	}
+
+	private void uploadFile(long fileEntryId, long commerceChannelId)
+		throws PortalException {
+		CommerceChannel commerceChannel =
+			_commerceChannelService.getCommerceChannel(
+				commerceChannelId);
+
+		FileEntry currentTemplate =
```
Comment (ivicac) [inline modules/apps/commerce/commerce-channel-web/src/main/java/com/liferay/commerce/channel/web/internal/portlet/action/EditCommerceChannelMVCActionCommand.java:133]: @christian-chiappa can you please use FileEntry as a suffix?

File: modules/apps/commerce/commerce-channel-web/src/main/java/com/liferay/commerce/channel/web/internal/portlet/action/EditCommerceChannelMVCActionCommand.java [hunk 3]
```diff
@@ -81,15 +92,91 @@ protected void doProcessAction(
...
+			_dlAppLocalService.fetchFileEntryByExternalReferenceCode(
+				commerceChannel.getGroupId(), "PRINT_ORDER");
+
+		if ((fileEntryId == 0) && (currentTemplate != null)) {
+			_dlAppLocalService.deleteFileEntry(
+				currentTemplate.getFileEntryId());
+		}
+		else {
+			FileEntry newTemplate = _dlAppLocalService.getFileEntry(
```
Comment (ivicac) [inline modules/apps/commerce/commerce-channel-web/src/main/java/com/liferay/commerce/channel/web/internal/portlet/action/EditCommerceChannelMVCActionCommand.java:142]: @christian-chiappa can you please use FileEntry as a suffix?

File: modules/apps/commerce/commerce-channel-web/src/main/java/com/liferay/commerce/channel/web/internal/portlet/action/EditCommerceChannelMVCActionCommand.java [hunk 4]
```diff
@@ -81,15 +92,91 @@ protected void doProcessAction(
...
+				fileEntryId);
+
+			if (!Objects.equals(newTemplate.getExtension(), "jrxml")) {
+				throw new FileExtensionException();
+			}
+
+			try {
+				if (currentTemplate != null) {
```
Comment (ivicac) [inline modules/apps/commerce/commerce-channel-web/src/main/java/com/liferay/commerce/channel/web/internal/portlet/action/EditCommerceChannelMVCActionCommand.java:150]: @christian-chiappa switch logic, go first with if (currentTemplate == null) {

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
ci:test:relevant

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
ci:test:relevant

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
Resent at https://github.com/liferay-commerce/liferay-portal/pull/2194

**Pull request closed without merging**